File: trunk/ShadowFrameworkJS/src/shadow/pipeline/openGL20/images/SFGL20RenderBuffer.js


function SFGL20RenderBuffer(width,height,format){
	this.width=width;
	this.height=height;
	this.format=format;
	this.renderBuffer=-1;
}


inherit(SFGL20RenderBuffer,SFBufferData);


SFGL20ListData.prototype["build"]=function(){
	if (renderBuffer == -1) {
		throw "Not yet translated";
		/*
		int rbo[] = new int[1];
		SFGL2.getGL().glGenRenderbuffers(1, rbo, 0);
		renderBuffer = rbo[0];
		SFGL2.getGL().glRenderbufferStorage(GL.GL_RENDERBUFFER,
				SFGL20RenderedTextureFactory.getGLFormat(getFormat()),
				getWidth(), getHeight());*/
	}
};


SFGL20ListData.prototype["destroy"]=function(){
	throw "Not yet implemented";
	/*
	int[] textures = { renderBuffer };
	SFGL2.getGL().glDeleteRenderbuffers(GL.GL_TEXTURE_2D, textures, 1);*/
};

File: ShadowFrameworkAndroid/src/com/shadow/android/opengles20/images/SFGL20RenderBuffer.java
package com.shadow.android.opengles20.images;

import javax.microedition.khronos.opengles.GL;

import shadow.image.SFBufferData;
import shadow.image.SFImageFormat;

import android.opengl.GLES20;

import com.shadow.android.opengles20.SFGL2;

/**
 * An object wrapping most of RenderBuffers functionalities
 * 
 * @author Alessandro
 */
public class SFGL20RenderBuffer extends SFBufferData implements
		SFGL20ImageObject {

	int renderBuffer = -1;

	public SFGL20RenderBuffer(int width, int height, SFImageFormat format) {
		super(width, height, format);
	}

	/*
	 * (non-Javadoc)
	 * 
	 * @see shadow.pipeline.openGL20.images.SFGL20ImageObject#build()
	 */
	@Override
	public void build() {

		if (renderBuffer == -1) {
			int rbo[] = new int[1];
			
			
			GLES20.glGenRenderbuffers(1, rbo, 0);
			renderBuffer = rbo[0];
			GLES20.glBindRenderbuffer(GLES20.GL_RENDERBUFFER, rbo[0]);
			GLES20.glRenderbufferStorage(GLES20.GL_RENDERBUFFER,
					GLES20.GL_DEPTH_COMPONENT16,
					//SFGL20RenderedTextureFactory.getGLFormat(getFormat()),
					getWidth(), getHeight());
			GLES20.glBindRenderbuffer(GLES20.GL_RENDERBUFFER, 0);
		}
	}

	/*
	 * (non-Javadoc)
	 * 
	 * @see shadow.pipeline.openGL20.images.SFGL20ImageObject#destroy()
	 */
	@Override
	public void destroy() {
		int[] textures = { renderBuffer };
		GLES20.glDeleteRenderbuffers(GLES20.GL_TEXTURE_2D, textures, 1);
	}
}
